pkg/proxy/topology.go

```diff
@@ -68,6 +68,9 @@ func filterEndpointsWithHints(endpoints []Endpoint, hintsAnnotation string, node
 	filteredEndpoints := []Endpoint{}
 
 	for _, endpoint := range endpoints {
+		if !endpoint.IsReady() {
+			continue
+		}
```

Comment:
do we need to do the same in the loop of L96 in this file?

Reply:
I don't think so. TrafficPolicy=Local doesn't have a similar failover mechanism, if there are no local endpoints we shouldn't fall back to all endpoints like we're doing here. Also with the related work around terminating endpoints, we will send traffic to non-ready terminating local endpoints if no others are available.

Reply:
ic, :+1: